HPXMLtoOpenStudio/measure.rb

```diff
@@ -1773,27 +1777,31 @@ def self.add_residual_ideal_system(runner, model, spaces)
     # the HVAC systems are undersized to meet the load). This allows us to correctly calculate total
     # heating/cooling loads without having to run an additional EnergyPlus simulation solely for that purpose,
     # as well as allows us to report the unmet load (i.e., the energy transferred by this ideal air system).
+    return if @hpxml.hvac_controls.empty? # no hvac system
 
     living_zone = spaces[HPXML::LocationLivingSpace].thermalZone.get
     obj_name = Constants.ObjectNameIdealAirSystemResidual
 
-    if @remaining_cool_load_frac < 1.0
-      sequential_cool_load_frac = 1.0
-    else
-      sequential_cool_load_frac = 0.0 # no cooling system, don't add ideal air for cooling either
-    end
+    # only enable residual ideal system when hvac's not on on/off thermostat control
```

Comment:
If the on off thermostat feature is enabled, it is not compatible with ideal system approach. 
1. For residual ideal system, it will perfectly resolve all remaining loads at each timestep, resulting in very short cycle (1 timestep per cycle regardless of hvac system capacity), so the residual ideal system should be disabled. 
2. For ideal systems that created for partial hvac, I verified that the on-off thermostat will also work oddly (short cycle). Taking an example of hvac system with on-off thermostat serving 33% of loads, for the current time step, ideal system first met 67% loads, and the 33% loads are assigned to hvac. If this 33% is not met perfectly, then the loads will be carried over to the next time step, and being resolved mostly by ideal system (67%). So ideal system actually helps resolve the current loads assigned to actual hvac system in the next time step.

Considering this situation, I think our feature is more reasonable to only be enabled in fully conditioned houses. @jmaguire1 @shorowit

Reply:
@yzhou601: I agree, I think we should only be using this feature for homes where 100% of the load is met by a single HVAC system. I think we at least want to throw a warning (probably an error) in this scenario, with a message letting the user know why we didn't allow this and what they need to change (either remove the deadband thermostat or fully condition the space with a single HVAC system)